dotCMS/src/main/java/com/dotcms/storage/ContentletMetadataAPIImpl.java

```diff
@@ -0,0 +1,255 @@
+                                    .full(false)
+                                    .override(alwaysRegenerateMetadata)
+                                    .cache(()-> ContentletCache.META_DATA_MAP_KEY + contentlet.getInode())
+                                    .metaDataKeyFilter(metadataKey -> metadataFields.isEmpty() ? true : metadataFields.contains(metadataKey))
+                                    .storageKey(new StorageKey.Builder().bucket(metadataBucketName).path(metadataPath).storage(storageType).build())
+                                    .build()
+                    );
+                }
+
+                basicMetadataMap.put(basicBinaryFieldName, metadataMap);
+            } else {
+
+                throw new IOException("The file: " + file + ", is null, does not exists or can not access");
+            }
+        }
+    }
+
+    private String getFileName (final Contentlet contentlet, final String fieldVariableName) {
+
+        final String inode        = contentlet.getInode();
+        final String fileName     = fieldVariableName + "-metadata.json";
+        return StringUtils.builder(File.separator,
+                inode.charAt(0), File.separator, inode.charAt(1), File.separator, inode, File.separator,
+                fileName).toString();
+    }
+
+    private void generateFullMetadata(final Contentlet contentlet,
+                                      final Set<String> fullBinaryFieldNameSet,
+                                      final ImmutableMap.Builder<String, Map<String, Object>> fullMetadataMap,
+                                      final Map<String, Field> fieldMap,
+                                      final boolean alwaysRegenerateMetadata) throws IOException {
+
+        final String storageType        = Config.getStringProperty("DEFAULT_STORAGE_TYPE", StorageType.FILE_SYSTEM.name());
+        Map<String, Object> metadataMap = Collections.emptyMap();
+        final String metadataBucketName = Config.getStringProperty("METADATA_BUCKET_NAME", "dotmetadata");
+        for (final String fullBinaryFieldName : fullBinaryFieldNameSet) {
+
+            final File file           = contentlet.getBinary(fullBinaryFieldName);
+            final String metadataPath = this.getFileName(contentlet, fullBinaryFieldName);
+            if (null != file && file.exists() && file.canRead()) {
+
+                final Set<String> metadataFields = this.getMetadataFields(fieldMap.get(fullBinaryFieldName).id());
+                metadataMap = this.fileStorageAPI.generateMetaData(file,
+                        new GenerateMetaDataConfiguration.Builder()
+                            .full(true)
+                            .override(alwaysRegenerateMetadata)
+                            .cache(false)  // do not want cache on full meta
+                            .metaDataKeyFilter(metadataKey -> metadataFields.isEmpty()? true: metadataFields.contains(metadataKey))
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Ternary operators that can be simplified with || or &&](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5547873)